dags/general/monitoring.py

```diff
@@ -0,0 +1,133 @@
+from airflow import DAG
+import json
+from operators.mattermost_operator import MattermostOperator
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.http_operator import SimpleHttpOperator
+from datetime import datetime
+from airflow_utils import send_alert
+from airflow.models.xcom import XCom
+from airflow.utils.db import provide_session
+from airflow.models import Variable
+import logging
+task_logger = logging.getLogger('airflow.task')
+
+
+"""
+This method returns list of extractions that are failing
+tap_exit_status = 1, failing
+tap_exit_status = 0, success
+tap_exit_status = None, extraction in process
+"""
+def stitch_check_extractions(response):
+    task_logger.info('Got extractions response from stitch api, checking for errors')
+    failed_extractions={}
+    try:
+        extractions = json.loads(response)['data']
+        for extraction in extractions:
+            if extraction['tap_exit_status'] is 1:
+                failed_extractions[extraction['source_id']] = extraction['tap_description']
+    except Exception as e:
+        task_logger.error('error in getting extractions from stitch ', str(e))
+        raise(e)
```

Comment:
Log and reraise is generally avoided. One of the main reasons is that it's easy for the exception to be logged elsewhere, wasting resources.

[This article](https://www.loggly.com/blog/exceptional-logging-of-exceptions-in-python/) describes some of the most common practices for handling exceptions.